dotenv-linter/tests/common/test_dir.rs

```diff
@@ -14,20 +15,32 @@ use std::str::from_utf8;
 /// Use to test commands in temporary directories
 pub struct TestDir {
     current_dir: TempDir,
+    envs: Option<HashMap<String, String>>,
 }
 
 impl TestDir {
     /// Generate a new TestDir
     pub fn new() -> Self {
         Self {
             current_dir: tempdir().expect("create testdir"),
+            envs: None,
+        }
+    }
+
+    // Only used in tests
+    #[allow(dead_code)]
```

Comment:
Do we really need it? 🤔 `new` also used only in tests AFAIR. Usage exists, so it should be a dead code

Reply:
I think it may have something to do with the test being part of a specific features.

Running `cargo t`, a warning will be logged about this:
```
cargo t
   Compiling dotenv-linter v3.3.0 (/home/jackyef/Projects/dotenv-linter/dotenv-linter)
warning: associated function `with_envs` is never used
  --> dotenv-linter/tests/common/test_dir.rs:30:12
   |
30 |     pub fn with_envs(envs: HashMap<String, String>) -> Self {
   |            ^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

^[[Awarning: `dotenv-linter` (test "cli") generated 1 warning
```

The warning won't appear when running `cargo t --all-features` though! Is it preferable to remove this anyway?

Reply:
Usually we use `all-features` ([CI workflow](https://github.com/dotenv-linter/dotenv-linter/blob/master/.github/workflows/ci.yml)), however you can leave it as it is. I don't mind